src/hotspot/cpu/aarch64/c1_LIRGenerator_aarch64.cpp

```diff
@@ -234,7 +234,8 @@ LIR_Opr LIRGenerator::load_immediate(int x, BasicType type) {
       return tmp;
     }
   } else if (type == T_INT) {
-    r = LIR_OprFact::intConst(x);
+    assert(min_jint <= x && x <= max_jint, "in range");
+    r = LIR_OprFact::intConst((jint)x);
```

Comment:
Hi, you could use `checked_cast`, which resides in `globalDefinitions.hpp` for this.

Reply:
Good suggestion, fixed!